f3write.c

```diff
@@ -184,6 +191,11 @@ static int create_and_fill_file(const char *path, long number, size_t size,
 	/* Create the file. */
 	full_fn = full_fn_from_number(&filename, path, number);
 	assert(full_fn);
+	if(access(full_fn, F_OK) != -1 && keep)
+	{
+		printf("Skipping file %s\n", filename);
+		return false;
+	}
```

Comment:
Instead of adding this `if` statement here, do as follows:

1. Replace `O_TRUNC` with `O_EXCL` in `open()` below;
2. Inside of the body of `if (fd < 0) {` add the following code:
```
if (errno == EEXIST) {
	if (keep) {
		printf("Skipped.\n");
		free(full_fn);
		return false;
	}
	err(errno, "Unexpectedly found file %s, but option --keep is not in use", full_fn);
}
```